src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -355,8 +375,29 @@ public Visitable visit(ClassOrInterfaceDeclaration node, Void arg) {
           this.gotException = true;
         }
         addedTargetFiles.add(filePath);
+      } else {
+        UnsolvedClassOrInterface unsolvedInterface;
+        try {
+          implementedOrExtended.resolve();
+          continue;
+        } catch (UnsolvedSymbolException e) {
+          unsolvedInterface = new UnsolvedClassOrInterface(typeName, packageName);
+        }
+        // this extended/implemented type is an interface if it is in the declaration of an
+        // interface, or if it is used with the "implements" keyword.
+        boolean typeIsAnInterface =
+            node.isInterface() || implementedTypes.contains(implementedOrExtended);
+        if (typeIsAnInterface) {
+          unsolvedInterface.setIsAnInterface();
+          classToItsUnsolvedInterface.put(className, typeName);
+        }
+        updateMissingClass(unsolvedInterface);
       }
     }
+
+    addTypeVariableScope(node.getTypeParameters());
+    Visitable result = super.visit(node, arg);
+    typeVariables.removeFirst();
```

Comment:
Why was this change necessary? Did we just miss this before? If so, I'd expect a test for it (and possibly a separate PR, since it seems unrelated to the bug at hand).

Reply:
These three lines have already existed. I simply changed their position. Previously, they are placed before the inheritance-handling loop. For this PR, we create new synthetic interfaces/superclasses during inheritance-handling loop, so we need to put these three lines after the loop in order to have JavaParser take the newly created synthetic classes into consideration when visiting other nodes.

Reply:
Thanks for the explanation - sorry I missed the corresponding deletions.